baseline-network-linux-hld.md

```diff
@@ -0,0 +1,162 @@
+# Unified Open Networking Hardware Support
+
+[[_TOC_]]
+
+## Scope
+
+The current open network operating system (NOS) market is fragmented and has
+multiple competing projects with different focus, architecture and level of
+maturity. Despite those differences, all of the currently available,
+open network operating systems are utilizing a mainline Linux kernel and
+need to build and maintain platform-specific abstractions (e.g., out-of-tree
+drivers) for peripheral switch hardware.
+
+This proposal aims to reduce the fragmentation by providing a common base for
+networking hardware support, without addressing the question of how to abstract
+and control the switch forwarding plane itself.
+
+The intention of this project is not to replace or compete with any of the
+existing network operating systems, but rather to provide a common and
+extensible layer on which they can be built. Its principles and paradigms are
+largely aligned with and adapted from the
+[OCP OpenNetworkLinux Project](https://github.com/opencomputeproject/OpenNetworkLinux) (which has been moved to maintenance mode, pending archival).
+This proposal specifically seeks an agreement between the developers and
+maintainers of the aforementioned network operating system communities, to be
+able to share common implementations to support open networking hardware.
+
+Support for open networking hardware platforms is currently distributed over
+various different projects. Many platforms, especially older ones, are supported
+by OpenNetworkLinux, many are supported by SONiC, and some are supported by
+DentOS, which is based on OpenNetworkLinux, but targets different devices. The
+open hardware support is usually very tightly integrated in the build systems,
+which makes sharing and reusing the support hard.
+
+## Requirements and Dependencies
+
+## Functional Description
+
+### Base Open Networking Hardware Support
+
+The base networking hardware support combines hardware support from multiple
+sources, and therefore will need to support multiple kinds of APIs.
+
+Possible APIs are:
+
+- SONiC PDDF, due to the large amount of available platforms
+  https://github.com/sonic-net/SONiC/blob/master/doc/platform/brcm_pdk_pddf.md
+- S3IP as described in https://github.com/sonic-net/SONiC/pull/1068
+- ONLP, as used by ONL and DentOS
+
+The base networking hardware support is organized in multiple repositories:
+
+- kernel: Linux fork with required base changes (backports of important
+submissions, etc)
+- platform-support: drivers, libraries and userspace code for initializing
+platforms
+(do we need more?)
+- TBD: organization of platform support
+- TBD: platform initialization
+
+In the first iteration, the platform support is fine as-is. Future iterations
```

Comment:
What will be the first platform supported?

Reply:
I think we discussed the EdgeCore AS 4630 in previous calls as a widely adapted and pretty useful one, so we might want to make this one our "demo platform". Since we are focussing on a BSP Layer here, it might be worth evaluating how many "low hanging fruits" from the ONL and SONiC side we can find and then add them in one go. In addition to the current code status, we might want to continue the discussion with whitebox vendors to see what they think and what would be useful for them.